src/context.rs

```diff
@@ -158,6 +162,57 @@ impl Context {
         self.blobdir.as_path()
     }
 
+    pub fn copy_to_blobdir(&self, orig_filename: impl AsRef<str>) -> Result<String> {
+        // return a $BLOBDIR/<filename> with the content of orig_filename
+        // copied into it. The <filename> will be safely derived from
+        // orig_filename, and will not clash with existing filenames.
```

Comment:
Please write doc comments according to https://doc.rust-lang.org/rust-by-example/meta/doc.html and check them using `cargo doc --no-deps``